apps/src/templates/progress/BubbleFactory.jsx

```diff
@@ -82,15 +91,43 @@ DiamondContainer.propTypes = {
   children: PropTypes.node
 };
 
-export function LinkWrapper({url, children}) {
+export function BubbleLink({url, onClick, children}) {
   return (
-    <a href={url} style={bubbleStyles.link}>
+    <a href={url} onClick={onClick} className="progress-bubble-link">
       {children}
     </a>
   );
 }
-LinkWrapper.propTypes = {
-  url: PropTypes.string.isRequired,
+BubbleLink.propTypes = {
+  url: PropTypes.string,
+  onClick: PropTypes.func,
+  children: PropTypes.element.isRequired
+};
+
+export function BubbleTooltip({level, children}) {
+  let tooltipText = level.isSublevel
+    ? level.display_name
+    : level.isUnplugged
+    ? i18n.unpluggedActivity()
+    : level.name || level.progressionDisplayName || '';
+  if (level.levelNumber) {
+    tooltipText = `${level.levelNumber}. ${tooltipText}`;
+  }
+  const tooltipId = _.uniqueId();
```

Comment:
Not sure what our philosophy on lodash is, but lodash seems like a big dependency to bring in just for this function?

Reply:
we use lodash enough that i would guess it's loaded on most pages anyway. since we were previously importing it in `ProgressBubble` (this code was just moved over from there) i'd rather not get sidetracked with overthinking this right now. but i can at least update to only import `uniqueId` instead of the whole library. you cool with leaving it at that for now?